### ISSUE TITLE
Can not run migrations in digital ocean server

### ISSUE THREAD

**Issue by nnoel2626** [susanBuck/e15-spring20#33]
I keep getting this error message below when I run "php artisan migrate" in the production server:

```
SQLSTATE[HY000] [1045] Access denied for user 'e15nnoel'@'localhost' (using password: NO) (SQL  
  : select * from information_schema.tables where table_schema = wireless_DB and table_name = mi  
  grations and table_type = 'BASE TABLE') 
```
I checked and rechecked my username and password. This error happened  rather I used my root user credential or the new created user. Through terminal command, I can access my SQL table but I can access the database using the artisan commend. Any help will be appreciated!

**Comment by budostylz** [comment]
This looks like a permission issue where the application might not have access to the migrations folder or any related database folders. I would check to see if the app have permissions for any database related folders. Hope this helps.


[Permissions Link](https://hesweb.dev/e15/notes/command-line/permissions)

**Comment by susanBuck** [comment]
👍to @budostylz's notes and suggestions.

If that doesn't do the trick - you can share server access with me [via these instructions](https://hesweb.dev/e15/notes/appendix/instructor-server-access) and I can take a look.

**Comment by nnoel2626** [comment]
I found this Github link that explains how to solve this problem: 
[https://github.com/laravel/framework/issues/27742. ](url)

Thank you all for your help.

**Comment by susanBuck** [comment]
Glad you found a solution. : )

I'm wondering why others of us aren't running into the same problem, as the thread you linked to suggests it's a problem specific to a version of PHP.

Out of curiosity, the production server you're using - is it something you created new for this semester, or one you had from a previous semester? 

That could explain the discrepancy.

**Comment by nnoel2626** [comment]
I created the droplet when I started the e15 class. The only thing that is different, I am using the Laravel version 7  and  PHP 7.2 version. I think that might be what causes this issue.

**Comment by nnoel2626** [comment]
I also open a ticket with Digital Ocean. Here is one of the emails they sent:

```
Hi there,

Thanks for getting in touch with us! Our team will be following up as soon as possible to assist you! In the meantime, we would also like to share some information on Migrating your existing MySQL database to our managed MySQL database. At this moment, the MySQL managed database is MySQL 8,  if you have an existing MySQL database you likely have used mysqldump client.

Ref: https://dev.mysql.com/doc/refman/5.7/en/mysqldump.html

Setup managed MySQL database

You will need to set up a managed MySQL database

https://www.digitalocean.com/docs/databases/mysql/how-to/create/

https://www.digitalocean.com/docs/databases/mysql/quickstart/#create-mysql-database-clusters

Backup your database

On your existing database, use the mysqldump client to take a logical backup

Example:  mysqldump -u root -p --opt [database name] > [database name].sql

Import your logical backed up database

The below command is an example of how import you backed database to the managed MySQL database 

Example: mysql -u doadmin -p -h db-mysql-nyc1-XXXXX-do-user-XXXXXXX-0.db.ondigitalocean.com -P 25060 < dbexport.sql
```

**Comment by susanBuck** [comment]
> The only thing that is different, I am using the Laravel version 7 and PHP 7.2 version.

Out of curiosity, I just set up a database for a Laravel 7 example app I created for testing purposes, and confirmed my production server is also on PHP version 7.2. All went well (output below), so I'm still puzzled.

Regardless, the important part is you got it fixed. : )

FYI - No need to switch to a managed MySQL database for this course, as that reply form DO suggests you do. They're just trying to push more of their services. : ) 


Laravel 7 db test:
```
root@e15:/var/www/html/e15/laravel7-example# php artisan migrate
**************************************
*     Application In Production!     *
**************************************

 Do you really wish to run this command? (yes/no) [no]:
 > yes

Migration table created successfully.
Migrating: 2014_10_12_000000_create_users_table
Migrated:  2014_10_12_000000_create_users_table (0.01 seconds)
Migrating: 2019_08_19_000000_create_failed_jobs_table
Migrated:  2019_08_19_000000_create_failed_jobs_table (0.01 seconds)

root@e15:/var/www/html/e15/laravel7-example# php --version
PHP 7.2.28-3+ubuntu18.04.1+deb.sury.org+1 (cli) (built: Feb 23 2020 07:23:25) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.28-3+ubuntu18.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies
```

**Comment by susanBuck** [comment]
I'm going to close this issue as the problem is resolved, but feel free to keep the discussion going if you want to dig deeper.

**Comment by RGPhD** [comment]
Hi @nnoel2626 and Professor @susanBuck,

I am also using version Laravel 7 and PHP version 7.2 on my p3.  The only differences I have had so far are: 
*1.* I do not have a 2014_10_12_100000_create_password_resets_table.php in p3 migrations (I checked and other students have the same migration files as I do).   
*2.* My create websites table did not make the id field a ```bigIncrement```, so I did this myself when creating data for migrations.

Nothing has caused any errors yet 😻.

Robelyn
